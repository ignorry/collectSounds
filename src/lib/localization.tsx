import React, { ReactNode, useState, useEffect } from "react";
import { IntlProvider } from "react-intl";

export type Lang = 'en' | 'ru';
export type Page = 
  'search' |
  'saved' |
  'queue' |
  'settings' |
  'video' |
  'playlist' |
  'channel' |
  'synchronization';

/**
 * import specific translation file
 * @param {Page} page 
 * @param {Lang} lang 
 */
export const loadMessages = ( page: Page, lang: Lang ) => import( `../translations/${page}/${lang}.json` );

/**
 * get user's language
 * @returns {Lang} lang code (eg 'en', 'ru')
 */
export const getLang = (): Lang => {
  const code = navigator.language.slice( 0, 2 );

  if ( code === 'ru' ) return code;

  return 'en';
}

/**
 * @typedef Props
 * @prop {string} file - translation file
 * @prop {ReactNode} children
 */
export type Props = {
  file: Page;
  children?: ReactNode;
  skeleton?: ReactNode;
}

/**
 * wrap component with TranslationProvider
 * @param {Props} props
 */
export const LangProvider: React.FC<Props> = ( props: Props ) => {
  const locale = getLang();
  const [messages, setMessages] = useState( null );

  useEffect(() => {
    loadMessages( props.file, locale ).then( setMessages );
  }, [locale]);

  return messages ? (
    <IntlProvider locale={locale} messages={messages}>
      { props.children }
    </IntlProvider>
  ) : <> { props.skeleton } </>;
}