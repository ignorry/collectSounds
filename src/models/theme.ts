export enum ThemeEnum  {
  light = 'light',
  dark = 'dark'
}

// all sizes in rem
export type Theme = {
  iconSize: number,
  rangeSize: number,
  spinnerSize: number,
  contentWidth: number,
  colors: {
    main: string,
    bg?: string,
    bgSecondary?: string,
    font?: string,
    darkBg?: string,
  },
  decorative: { // in px
    borderRadius: number;
    padding: number;
  },
  gaps: {
    smallest: number,
    small: number,
    big: number,
  },
  thumbnail: {
    width: number,
    height: number,
    infoColor: string,
    infoWidth: 4,
  },
  error: {
    backgroundColor: string,
    width: number,
  },
  media: {
    small: string,
  },
  font: {
    primary: {
      size: number,
      opacity: number,
    },
    secondary: {
      size: number,
      opacity: number,
    },
  },
  durations: {
    ms300: number,
  },
  order: {
    footer: number,
    modal: number,
  },
}