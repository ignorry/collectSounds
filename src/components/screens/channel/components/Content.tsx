import React from "react";
import styled from "styled-components";
import { Channel } from "../../../../models/content";
import { useIntl } from "react-intl";
import { useNavigate } from "react-router-dom";
import queryString from "query-string";

import Label from "../../../primitives/Label";
import Button from "../../../primitives/Button";
import Spinner from "../../../primitives/Spinner";
import List from "../../../content/List";
import Tabs from "../../../primitives/Tabs";
import AddTag from "../../../primitives/AddTag";

const NODE_ROUTE = !process.env.NODE_ENV || process.env.NODE_ENV === 'development' ? 1: 2;
const BASE_URL = !process.env.NODE_ENV || process.env.NODE_ENV === 'development' ? "" : "/collectSounds/";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto;
`;

const Top = styled.div`
  background: ${ ({ theme }) => theme.colors.bgSecondary };
  padding-bottom: ${ ({ theme }) => `${ theme.font.primary.size*1.1 + theme.gaps.small*2 }rem`};
  margin-bottom: ${ ({ theme }) => `-${ theme.font.primary.size*1.1 + theme.gaps.small*2 }rem`};
`;

const Info = styled.div`
  background: ${ ({ theme }) => theme.colors.bgSecondary };
  padding: ${ ({ theme }) => `0 ${ theme.gaps.big }rem ${ theme.gaps.big }rem` };
  display: flex;
  flex-direction: column;
  gap: ${ ({ theme }) => `${ theme.gaps.big }rem` };
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  margin: auto;
`;

const Main = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: ${ ({ theme }) => `${ theme.gaps.small }rem` };
`;

const Left = styled.div`
  display: flex;
  align-items: center;
  gap: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const Img = styled.img`
  height: ${ ({ theme }) => `${ theme.thumbnail.height }rem` };
  border-radius: 50%;
  overflow: hidden;
`;

const TabsContainer = styled.div`
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  margin: auto;
`;

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

/**
 * @typedef Props
 * @prop {Function} toggleSaved - callback on click saved
 * @prop {boolean} saved - is video saved in store or not
 */
export type Props = {
  channel: Channel;
  toggleSaved: Function;
  saved: boolean;
}

const Content: React.FC<Props> = ( props: Props ) => {
  const navigate = useNavigate();
  const intl = useIntl();

  const tab = ( index: number ) => {
    const stringified = queryString.stringify( { index } );
    window.history.pushState('', '', `${ window.location.origin }${ window.location.pathname }?${ stringified }` );
  };

  const videos = props.channel.videos ? <List items={ props.channel.videos.map( item => ({
    content: item.item,
    callback: () => navigate( `${ BASE_URL || '/' }${ window.location.pathname.split( '/' )[NODE_ROUTE] }/video/${ item.id }` )
  })) } emptyMessage={ intl.formatMessage({ id: 'emptyVideos' }) }/> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  const playlists = props.channel.playlists ? <List items={ props.channel.playlists.map( item => ({
    content: item.item,
    callback: () => navigate( `${ BASE_URL || '/' }${ window.location.pathname.split( '/' )[NODE_ROUTE] }/playlist/${ item.id }` )
  })) } emptyMessage={ intl.formatMessage({ id: 'emptyPlaylists' }) }/> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  const tabs = [
    props.channel.description && props.channel.description.length > 0 && {
      label: intl.formatMessage({ id: 'about' }),
      content: <Label text={ props.channel.description } />
    },
    {
      label: intl.formatMessage({ id: 'videos' }),
      content: videos
    },
    {
      label: intl.formatMessage({ id: 'playlists' }),
      content: playlists
    }
  ];

  return (
    <Container>
      <Top>
        <Info>
          <Main>
            <Left>
                <Img
                  src={ props.channel.thumbnails.standard?.url || props.channel.thumbnails.default?.url || '' }
                />
              <Label text={ props.channel.title }/>
            </Left>
            <Button
              callback={ props.toggleSaved } pic="playlist"
              active={ props.saved }
            />
          </Main>
          <AddTag item={ props.channel }/>
        </Info>
      </Top>
      <TabsContainer>
        <Tabs tabs={ tabs } onChange={ tab } active={ +queryString.parse( location.search ).index || 0 }/>
      </TabsContainer>
    </Container>
  );
}

export default Content;