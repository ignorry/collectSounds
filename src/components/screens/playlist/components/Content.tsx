import React, { useEffect, useRef, useState } from "react";
import styled from "styled-components";
import { Playlist } from "../../../../models/content";
import { useIntl } from "react-intl";
import { SearchFilters } from "../../../../lib/api";
import { getDurationFromMs } from "../../../../lib/getDuration";

import Range from "../../../primitives/Range";
import Label from "../../../primitives/Label";
import PlayerButton from "../../../primitives/PlayerButton";
import Button from "../../../primitives/Button";
import Collapse from "../../../primitives/Collapse";
import Spinner from "../../../primitives/Spinner";
import List from "../../../content/List";
import { useNavigate } from "react-router-dom";
import AddTag from "../../../primitives/AddTag";

const NODE_ROUTE = !process.env.NODE_ENV || process.env.NODE_ENV === 'development' ? 1: 2;
const BASE_URL = !process.env.NODE_ENV || process.env.NODE_ENV === 'development' ? "" : "/collectSounds/";

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const Main = styled.div`
  background: ${ ({ theme }) => theme.colors.bgSecondary };
`;

const Column = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  padding: ${ ({ theme }) => `0 ${ theme.gaps.big }rem ${ theme.gaps.big }rem` };
  display: flex;
  flex-direction: column;
  gap: ${ ({ theme }) => `${ theme.gaps.small }rem` };
`;

const Line = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-top: ${ ({ theme }) => `-${ theme.gaps.small }rem` };
`;

const Description = styled.div`
  text-align: left;
`;

const Videos = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

/**
 * @typedef Props
 * @prop {Function} toggleSaved - callback on click saved
 * @prop {boolean} saved - is video saved in store or not
 */
export type Props = {
  playlist: Playlist;
  toggleSaved: Function;
  saved: boolean;
}

const Content: React.FC<Props> = ( props: Props ) => {
  const navigate = useNavigate();
  const intl = useIntl();

  const items = props.playlist.videos ? <List items={ props.playlist.videos.map( item => ({
    content: item.item,
    callback: () => navigate( `${ BASE_URL || '/' }${ window.location.pathname.split( '/' )[NODE_ROUTE] }/video/${ item.id }` )
  })) } emptyMessage={ intl.formatMessage({ id: 'emptyList' }) }/> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  return (
    <Container>
      <Main>
        <Column>
          <Label text={ props.playlist.title }/>
          <Line>
            <Label text={ props.playlist.channelTitle }/>
            <Button
              callback={ props.toggleSaved } pic="playlist"
              active={ props.saved }
            />
          </Line>
          <AddTag item={ props.playlist }/>
          { props.playlist.description && props.playlist.description.length > 0 ?
            <Collapse label={ intl.formatMessage({ id: 'description' }) }> 
              <Description>
                <Label text={ props.playlist.description }/>
              </Description>
            </Collapse> : null
          }
        </Column>
      </Main>
      <Videos>
        { items }
      </Videos>
    </Container>
  );
}

export default Content;