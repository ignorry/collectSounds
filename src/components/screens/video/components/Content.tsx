import React, { useEffect, useRef, useState } from "react";
import styled from "styled-components";
import { Video } from "../../../../models/content";
import { useIntl } from "react-intl";
import { SearchFilters } from "../../../../lib/api";
import { getDurationFromMs } from "../../../../lib/getDuration";

import Range from "../../../primitives/Range";
import Label from "../../../primitives/Label";
import PlayerButton from "../../../primitives/PlayerButton";
import Button from "../../../primitives/Button";
import AddTag from "../../../primitives/AddTag";

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const Main = styled.div`
  background: ${ ({ theme }) => theme.colors.bgSecondary };
`;

const Column = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const ThumbContainer = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  padding: ${ ({ theme }) => `0 ${ theme.gaps.big }rem` };
  
  @media ${ ({ theme }) => theme.media.small } {
    padding: 0;
  }
`;

const Thumbnail = styled.div<{ realWidth: number }>`
  height: ${ props => ({ theme }) => `${ theme.thumbnail.height/theme.thumbnail.width*props.realWidth }px` };
  width: 100%;
  overflow: hidden;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
`;

const Img = styled.img`
  width: 100%;
`;

const RangeContainer = styled.div`
  position: absolute;
  width: 100%;
  bottom: ${ ({ theme }) => `-${ theme.rangeSize }rem` };

  input {
    width: 100%;
  };
`;

const Play = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Length = styled.span`
  display: block;
  position: absolute;
  right: ${ ({ theme }) => `${ theme.gaps.small }rem` };
  bottom: ${ ({ theme }) => `${ theme.gaps.small }rem` };
  background: ${ ({ theme }) => theme.thumbnail.infoColor };
  padding: 0 ${ ({ theme }) => `${ theme.decorative.padding }px` };
  border-radius: ${ ({ theme }) => `${ theme.decorative.borderRadius }px` };
`;

const Info = styled.div`
  padding: ${ ({ theme }) => `0 ${ theme.gaps.big }rem ${ theme.gaps.big }rem` };
  display: flex;
  flex-direction: column;
  gap: ${ ({ theme }) => `${ theme.gaps.small }rem` };
`

const Line = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-top: ${ ({ theme }) => `-${ theme.gaps.small }rem` };
`;

const Description = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

/**
 * @typedef Props
 * @prop {Function} toggleSaved - callback on click saved
 * @prop {boolean} saved - is video saved in store or not
 */
export type Props = {
  video: Video;
  toggleSaved: Function;
  saved: boolean;
}

const Content: React.FC<Props> = ( props: Props ) => {
  const thumbnailRef: React.MutableRefObject<undefined | HTMLDivElement> = useRef();
  const [videoWidth, setVideoWidth] = useState<number>( 0 );
  
  useEffect( () => {
    setVideoWidth( thumbnailRef.current ? thumbnailRef.current.getBoundingClientRect().width : 0 );
  }, [thumbnailRef]);

  return (
    <Container>
      <Main>
        <Column>
          <ThumbContainer>
            <Thumbnail realWidth={ thumbnailRef.current ? thumbnailRef.current.getBoundingClientRect().width : 0 } ref={ thumbnailRef }>
              <Img
                src={ props.video.thumbnails.maxres?.url || props.video.thumbnails.default?.url || '' }
              />
              { props.video.passed &&
                <RangeContainer>
                  <Range value={ props.video.passed } max={ props.video.duration } />
                </RangeContainer>
              }
              <Play>
                <PlayerButton callback={() => {}} pic="play"/>
              </Play>
              <Length>
                <Label text={ getDurationFromMs( props.video.duration ) } secondary/>
              </Length>
            </Thumbnail>
          </ThumbContainer>
          <Info>
            <Label text={ props.video.title }/>
            <Label text={ props.video.channelTitle }/>
            <Line>
              <Label text={ props.video.publishedAt.replace( /[A-Z]/g, ' ' ) } secondary/>
              <Button
                callback={ props.toggleSaved } pic="playlist"
                active={ props.saved }
              />
            </Line>
            <AddTag item={ props.video }/>
          </Info>
        </Column>
      </Main>
      <Description>
        <Label text={ props.video.description }/>
      </Description>
    </Container>
  );
}

export default Content;