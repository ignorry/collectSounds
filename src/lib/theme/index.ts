import { DefaultTheme } from "styled-components";
import { ThemeEnum } from "../../models/theme";
import { baseTheme } from "./baseTheme";

export const lightTheme: DefaultTheme = {
  ...baseTheme,
  type: ThemeEnum.light,

  colors: {
    ...baseTheme.colors,
    main: '#00bb88',
    bg: '#ffffff',
    bgSecondary: '#d8d8d8',
    font: '#000000',
    darkBg: '#d9d9d9'
  },
}

export const darkTheme: DefaultTheme = {
  ...baseTheme,
  type: ThemeEnum.dark,

  colors: {
    ...baseTheme.colors,
    bg: '#000000',
    bgSecondary: '#2a2a2a',
    font: '#ffffff',
    darkBg: '#121212'
  },
}

/**
 * get theme depending on isDark parameter
 * @param {boolean} isDark 
 * @returns {DefaultTheme}
 */
export const getTheme = ( isDark: boolean ): DefaultTheme => isDark ? darkTheme : lightTheme;