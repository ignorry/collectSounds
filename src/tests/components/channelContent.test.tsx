import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { getTheme } from "../../lib/theme";
import GlobalStyles from "../../lib/styled/global";
import { Provider } from "react-redux";
import store from "../../redux/index";
import { LangProvider } from "../../lib/localization";
import { BrowserRouter } from "react-router-dom";
import { Channel, Playlist, Video } from "../../models/content";

import Content from "../../components/screens/channel/components/Content";
import userEvent from "@testing-library/user-event";

const AllTheProviders = ( { children }: any ) => (
  <Provider store={store}>
    <ThemeProvider theme={ getTheme( true ) } >
      <BrowserRouter>
        <LangProvider file="channel">
          { children }
        </LangProvider>
      </BrowserRouter>
      <GlobalStyles/>
    </ThemeProvider>
  </Provider>
);

const channel = {
  type: 'channel',
  id: 'someid',
  videos: [{
    id: 'video',
    item: {
      type: 'video',
      id: 'videoid',
      publishedAt: 'published at',
      title: 'video title',
      description: 'description',
      channelTitle: 'channel title',
      thumbnails: {
        standard: {
          url: 'some url',
        }
      }
    } as Video 
  }],
  playlists: [{
    id: 'playlistid',
    item: {
      type: 'playlist',
      id: 'someid',
      title: 'playlist title',
      description: 'description',
      channelTitle: 'channel title',
      thumbnails: {
        standard: {
          url: 'some url',
        }
      },
      videos: []
    } as Playlist
  }],
  title: 'title',
  description: 'description',
  customUrl: 'customUrl',
  thumbnails: {
    standard: {
      url: 'some url',
    }
  },
  lastModified: 0,
} as Channel;

describe( 'Content', () => {
  it( 'renders correctly', async () => {
    render(
      <Content channel={ channel } toggleSaved={ () => {} } saved={ false }/>,
      { wrapper: AllTheProviders }
    );

    await waitFor( () => {
      expect( screen.getByText( 'title' ) ).toBeTruthy();
    });
  });

  it( 'handles click on save button', async () => {
    const channel = { 
      type: 'channel',
      id: 'someid',
      videos: [],
      playlists: [],
      title: 'redux title',
      description: 'description',
      customUrl: 'customUrl',
      thumbnails: {
        default: {
          url: 'some url',
        }
      }
    } as Channel;

    const callback = jest.fn();

    const result = render(
      <Content channel={ channel } toggleSaved={ callback } saved={ false }/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    result.container.querySelector( 'button' ).click();

    expect( callback ).toBeCalled();
  });

  it( 'handles click on video', async () => {
    const user = userEvent.setup();

    const result = render(
      <Content channel={ channel } toggleSaved={() => {}} saved={ false }/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    await user.click( screen.getByText( 'video title' ) );

    expect( window.location.pathname ).toBe( '/collectSounds/undefined/video/video' );
  });

  it( 'handles click on playlist', async () => {
    const user = userEvent.setup();

    const result = render(
      <Content channel={ channel } toggleSaved={() => {}} saved={ false }/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    await user.click( screen.getByText( 'playlist title' ) );

    expect( window.location.pathname ).toBe( '/collectSounds/undefined/playlist/playlistid' );
  });

  it( 'matches to snapshot', async () => {

    const result = render(
      <Content channel={ channel } toggleSaved={ () => {} } saved={ false }/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    expect( result ).toMatchSnapshot();
  });
});