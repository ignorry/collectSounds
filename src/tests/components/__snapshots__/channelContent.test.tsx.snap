// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content matches to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iAvgwm daJpBG"
      >
        <div
          class="sc-efBctP edPByx"
        >
          <div
            class="sc-cTQhss caNHKO"
          >
            <div
              class="sc-jOrMOR cYkGWB"
            >
              <div
                class="sc-dPyBCJ eDKTuI"
              >
                <img
                  class="sc-bBXxYQ cFgMxI"
                  src="some url"
                />
                <p
                  class="sc-bczRLJ cTDXTp"
                >
                  title
                </p>
              </div>
              <button
                class="sc-dkzDqf fcjXBB"
              >
                <svg
                  class="sc-gsnTZi gSmFuM"
                >
                  <use
                    href="http://localhost/collectSounds/icons.svg#playlist"
                  />
                </svg>
              </button>
            </div>
            <div
              class="sc-gXmSlM gTGlkc"
            >
              <div
                class="sc-cCsOjp eRiddm"
              >
                <div
                  class="sc-ciZhAO lhNMbE"
                >
                  <p
                    class="sc-bczRLJ cTDXTp"
                  >
                    #
                  </p>
                  <div
                    class="sc-jdAMXn klNkfz"
                  />
                </div>
                <button
                  class="sc-bZkfAO eeVatE"
                >
                  <svg
                    class="sc-gsnTZi gSmFuM"
                  >
                    <use
                      href="http://localhost/collectSounds/icons.svg#add"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="sc-bZnhIo gMNbPu"
              >
                <div
                  class="sc-jdAMXn klNkfz"
                />
                <div
                  class="sc-iTONeN hFSBBE"
                >
                  <input
                    class="sc-jIZahH jGyggs"
                    placeholder="Add new"
                    value=""
                  />
                  <button
                    class="sc-dkzDqf fcjXBB"
                  >
                    <svg
                      class="sc-gsnTZi gSmFuM"
                    >
                      <use
                        href="http://localhost/collectSounds/icons.svg#add"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-cOFTSb gnjjyM"
        >
          <div>
            <div
              class="sc-kLLXSd dJACWh"
            >
              <button
                class="sc-bZkfAO htIohz"
              >
                <p
                  class="sc-bczRLJ cTDXTp"
                >
                  About
                </p>
              </button>
              <button
                class="sc-bZkfAO eeVatE"
              >
                <p
                  class="sc-bczRLJ cTDXTp"
                >
                  Videos
                </p>
              </button>
              <button
                class="sc-bZkfAO eeVatE"
              >
                <p
                  class="sc-bczRLJ cTDXTp"
                >
                  Playlists
                </p>
              </button>
            </div>
            <div
              class="sc-ikZpkk bZigCe"
            >
              <p
                class="sc-bczRLJ cTDXTp"
              >
                description
              </p>
            </div>
            <div
              class="sc-ikZpkk jHenAF"
            >
              <div
                class="sc-gicCDI krBTcR"
              >
                <div
                  class="sc-ivTmOn lGvzm"
                >
                  <div
                    class="sc-cxabCf hUgmmj"
                  >
                    <p
                      class="sc-bczRLJ cTDXTp"
                    >
                      save
                    </p>
                  </div>
                  <div
                    class="sc-iIPllB jdDgTD"
                  >
                    <button
                      class="sc-ftvSup fgTGsh"
                    >
                      <div
                        class="sc-papXJ jzECTc"
                      >
                        <img
                          class="sc-jqUVSM htJVeA"
                          src="some url"
                        />
                        <span
                          class="sc-iqcoie kWnFou"
                        >
                          <p
                            class="sc-bczRLJ cygevi"
                          >
                            NaN
                          </p>
                        </span>
                      </div>
                      <div
                        class="sc-crXcEl dnNpxW"
                      >
                        <div
                          class="sc-evZas bbzNcv"
                        >
                          <p
                            class="sc-bczRLJ cTDXTp"
                          >
                            video title
                          </p>
                        </div>
                        <div
                          class="sc-breuTD bppYZD"
                        >
                          <p
                            class="sc-bczRLJ cygevi"
                          >
                            channel title
                          </p>
                        </div>
                        <p
                          class="sc-bczRLJ cygevi"
                        >
                          published at
                        </p>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-ikZpkk jHenAF"
            >
              <div
                class="sc-gicCDI krBTcR"
              >
                <div
                  class="sc-ivTmOn lGvzm"
                >
                  <div
                    class="sc-cxabCf hUgmmj"
                  >
                    <p
                      class="sc-bczRLJ cTDXTp"
                    >
                      save
                    </p>
                  </div>
                  <div
                    class="sc-iIPllB jdDgTD"
                  >
                    <button
                      class="sc-ksZaOG eaOmpk"
                    >
                      <div
                        class="sc-hAZoDl kHPNAP"
                      >
                        <img
                          class="sc-fnykZs eWLXfB"
                          src="some url"
                        />
                        <span
                          class="sc-fEOsli exrbZj"
                        >
                          <p
                            class="sc-bczRLJ cTDXTp"
                          />
                          <svg
                            class="sc-gsnTZi gSmFuM"
                          >
                            <use
                              href="http://localhost/collectSounds/icons.svg#playlistThumbnail"
                            />
                          </svg>
                        </span>
                      </div>
                      <div
                        class="sc-bjUoiL djMEHe"
                      >
                        <div
                          class="sc-idiyUo bUPSPD"
                        >
                          <p
                            class="sc-bczRLJ cTDXTp"
                          >
                            playlist title
                          </p>
                        </div>
                        <div
                          class="sc-dIouRR gSWUXk"
                        >
                          <p
                            class="sc-bczRLJ cygevi"
                          >
                            channel title
                          </p>
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iAvgwm daJpBG"
    >
      <div
        class="sc-efBctP edPByx"
      >
        <div
          class="sc-cTQhss caNHKO"
        >
          <div
            class="sc-jOrMOR cYkGWB"
          >
            <div
              class="sc-dPyBCJ eDKTuI"
            >
              <img
                class="sc-bBXxYQ cFgMxI"
                src="some url"
              />
              <p
                class="sc-bczRLJ cTDXTp"
              >
                title
              </p>
            </div>
            <button
              class="sc-dkzDqf fcjXBB"
            >
              <svg
                class="sc-gsnTZi gSmFuM"
              >
                <use
                  href="http://localhost/collectSounds/icons.svg#playlist"
                />
              </svg>
            </button>
          </div>
          <div
            class="sc-gXmSlM gTGlkc"
          >
            <div
              class="sc-cCsOjp eRiddm"
            >
              <div
                class="sc-ciZhAO lhNMbE"
              >
                <p
                  class="sc-bczRLJ cTDXTp"
                >
                  #
                </p>
                <div
                  class="sc-jdAMXn klNkfz"
                />
              </div>
              <button
                class="sc-bZkfAO eeVatE"
              >
                <svg
                  class="sc-gsnTZi gSmFuM"
                >
                  <use
                    href="http://localhost/collectSounds/icons.svg#add"
                  />
                </svg>
              </button>
            </div>
            <div
              class="sc-bZnhIo gMNbPu"
            >
              <div
                class="sc-jdAMXn klNkfz"
              />
              <div
                class="sc-iTONeN hFSBBE"
              >
                <input
                  class="sc-jIZahH jGyggs"
                  placeholder="Add new"
                  value=""
                />
                <button
                  class="sc-dkzDqf fcjXBB"
                >
                  <svg
                    class="sc-gsnTZi gSmFuM"
                  >
                    <use
                      href="http://localhost/collectSounds/icons.svg#add"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-cOFTSb gnjjyM"
      >
        <div>
          <div
            class="sc-kLLXSd dJACWh"
          >
            <button
              class="sc-bZkfAO htIohz"
            >
              <p
                class="sc-bczRLJ cTDXTp"
              >
                About
              </p>
            </button>
            <button
              class="sc-bZkfAO eeVatE"
            >
              <p
                class="sc-bczRLJ cTDXTp"
              >
                Videos
              </p>
            </button>
            <button
              class="sc-bZkfAO eeVatE"
            >
              <p
                class="sc-bczRLJ cTDXTp"
              >
                Playlists
              </p>
            </button>
          </div>
          <div
            class="sc-ikZpkk bZigCe"
          >
            <p
              class="sc-bczRLJ cTDXTp"
            >
              description
            </p>
          </div>
          <div
            class="sc-ikZpkk jHenAF"
          >
            <div
              class="sc-gicCDI krBTcR"
            >
              <div
                class="sc-ivTmOn lGvzm"
              >
                <div
                  class="sc-cxabCf hUgmmj"
                >
                  <p
                    class="sc-bczRLJ cTDXTp"
                  >
                    save
                  </p>
                </div>
                <div
                  class="sc-iIPllB jdDgTD"
                >
                  <button
                    class="sc-ftvSup fgTGsh"
                  >
                    <div
                      class="sc-papXJ jzECTc"
                    >
                      <img
                        class="sc-jqUVSM htJVeA"
                        src="some url"
                      />
                      <span
                        class="sc-iqcoie kWnFou"
                      >
                        <p
                          class="sc-bczRLJ cygevi"
                        >
                          NaN
                        </p>
                      </span>
                    </div>
                    <div
                      class="sc-crXcEl dnNpxW"
                    >
                      <div
                        class="sc-evZas bbzNcv"
                      >
                        <p
                          class="sc-bczRLJ cTDXTp"
                        >
                          video title
                        </p>
                      </div>
                      <div
                        class="sc-breuTD bppYZD"
                      >
                        <p
                          class="sc-bczRLJ cygevi"
                        >
                          channel title
                        </p>
                      </div>
                      <p
                        class="sc-bczRLJ cygevi"
                      >
                        published at
                      </p>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-ikZpkk jHenAF"
          >
            <div
              class="sc-gicCDI krBTcR"
            >
              <div
                class="sc-ivTmOn lGvzm"
              >
                <div
                  class="sc-cxabCf hUgmmj"
                >
                  <p
                    class="sc-bczRLJ cTDXTp"
                  >
                    save
                  </p>
                </div>
                <div
                  class="sc-iIPllB jdDgTD"
                >
                  <button
                    class="sc-ksZaOG eaOmpk"
                  >
                    <div
                      class="sc-hAZoDl kHPNAP"
                    >
                      <img
                        class="sc-fnykZs eWLXfB"
                        src="some url"
                      />
                      <span
                        class="sc-fEOsli exrbZj"
                      >
                        <p
                          class="sc-bczRLJ cTDXTp"
                        />
                        <svg
                          class="sc-gsnTZi gSmFuM"
                        >
                          <use
                            href="http://localhost/collectSounds/icons.svg#playlistThumbnail"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="sc-bjUoiL djMEHe"
                    >
                      <div
                        class="sc-idiyUo bUPSPD"
                      >
                        <p
                          class="sc-bczRLJ cTDXTp"
                        >
                          playlist title
                        </p>
                      </div>
                      <div
                        class="sc-dIouRR gSWUXk"
                      >
                        <p
                          class="sc-bczRLJ cygevi"
                        >
                          channel title
                        </p>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
